docs/data-cookbooks/tracking-files.md

```diff
@@ -0,0 +1,121 @@
+---
+id: tracking-files
+title: Tracking Ancillary Files
+hide_title: true
+---
+
+# Tracking Files
+
+## Contents
+
+* [Introduction](#introduction)
+* [File Types](#file-types)
+* [Collection Configuration](#collection-configuration)
+* [Publish to CMR](#publish-to-cmr)
+* [Common Use Cases](#common-use-cases)
+
+### Introduction
+
+This document covers setting up ingest to track primary and ancillary files under various file types, which will carry through to the CMR and granule record.
+Cumulus has a number of options for tracking files and publishing files to CMR under certain metadata elements or with specified file types.
+We will cover Cumulus file types, collection configuration, and effects on publishing to CMR.
+
+### File types
+
+Cumulus follows the Cloud Notification Mechanism (CNM) file type conventions. Under this schema, there are four data types:
+
+* `data`
+* `browse`
+* `metadata`
+* `qa`
+
+In Cumulus, these data types are mapped to the `Type` attribute on `RelatedURL`s for UMM-G metadata, or used to map
+resources to one of `OnlineAccessURL`, `OnlineResource` or `AssociatedBrowseImages` for ECHO10 XML metadata.
+
+### Collection Configuration
+
+File types for each file in a granule can be configured at the collection level as below:
+
+```json
+{
+  "name": "MOD09GQ",
+  "version": "006",
+  "dataType": "MOD09GQ",
+  "granuleId": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}$",
+  "granuleIdExtraction": "(MOD09GQ\\..*)(\\.hdf|\\.cmr|_ndvi\\.jpg)",
+  "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.hdf",
+  "files": [
+    {
+      "bucket": "protected",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.hdf$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.hdf",
+      "fileType": "data"
+    },
+    {
+      "bucket": "protected-2",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}\\.cmr\\.xml$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104.cmr.xml",
+      "fileType": "metadata"
+    },
+    {
+      "bucket": "public",
+      "regex": "^MOD09GQ\\.A[\\d]{7}\\.[\\S]{6}\\.006\\.[\\d]{13}_ndvi\\.jpg$",
+      "sampleFileName": "MOD09GQ.A2017025.h21v00.006.2017034065104_ndvi.jpg",
+      "fileType": "browse"
+    }
+  ]
+}
+```
+
+Files on a particular granule which match a given file specification will be receive the provided `fileType` in the granule record,
+which will then be used to inform CMR metadata updates in the publish step.
+
+### Publish to CMR
+
+When publishing granule metadata to CMR, the `PostToCmr` task will perform a few metadata updates in the area of URLs for its respective metadata schema.
```

Comment:
This is good content.  I'm wondering if we should include it (or a reference to it)  in the feature doc: http://localhost:3000/cumulus/docs/next/features/ancillary_metadata , or restructure that somehow.  🤔

Reply:
7af13a26a55c98b8df549842e36bdbfda48ca3c6